pypeit/reduce.py

```diff
@@ -664,6 +671,17 @@ def global_skysub(self, skymask=None, update_crmask=True, trim_edg=(3,3),
         # Mask objects using the skymask? If skymask has been set by objfinding, and masking is requested, then do so
         skymask_now = skymask if (skymask is not None) else np.ones_like(self.sciImg.image, dtype=bool)
 
+        # Allow for previous sky to better estimate ivar
```

Comment:
I think we should consider simply always doing two iterations here updating the noise and omitting the previous_sky keyword altogether. The point is that if there is object flux in the image, then by using the sky-model as the noise (rather than the image) you will be more likely to reject the pixels containing object flux which is a good thing.